### ISSUE TITLE
[Launch API Docs] Implement the search

### ISSUE THREAD

**Issue by kptdobe** [adobe/Adobe-Developer-Site#11]
Goal
Reproduce with Helix the current search implementation. 

Go to https://docs.adobelaunch.com/ and enter text in the search field: it opens a popup and shows the search results.
This is an explored pattern with some dynamicity we have never implemented until now.

cc @tripodsan @trieloff

**Comment by trieloff** [comment]
The existing site is using Algolia for search and Algolia has been on my list of yet-to-explore things.

Algolia has built a Jekyll plugin https://github.com/algolia/jekyll-algolia that is using their API to feed their index. I could imagine doing something similar in a bot. One thing we need to figure out is how to deal with strains in Algolia, i.e. do we keep one index per strain, or do we keep the strain as a metadata on the object in the index and filter by strain on retrieval.

**Comment by kptdobe** [comment]
I had a quick look and you can easily create indexes: https://www.algolia.com/doc/api-client/javascript/getting-started/?language=javascript#init-index - It is smart enough to use an existing one or create a new one based on the name you give it.

But maintaining the indexes has nothing to do with the strains I think, it is content related. We would need an index per content branch:

* Every time you change some content, the CI or the Bot triggers an update for each modified object (markdown page in our case) on the "content + branch name" index
*  The client (the rendered webpage) should use [instantsearch.js](https://community.algolia.com/instantsearch.js/v2/getting-started.html) to query Algolia and render the results
* instantsearch.js must only be configured on rendering time to use the correct index name, i.e. "content + branch name"... This is the place where you need to find the info from the strain...

Should not be too complicated to put in place, only main exploration is what do you index ? the full md page, sections...
GitBook seems to do it per "header + paragraph" in a md file. Check https://docs.adobelaunch.com/?q=test
But it is not accurate, the cache does not seem to be updated, I get non existing content blocks...

**Comment by kptdobe** [comment]
Yes, but tech doc does not have sections. Or at least, they should not be calculated per thematic break but per header. Is the sectionifier extensible to cover this ?

**Comment by trieloff** [comment]
> Is the sectionifier extensible to cover this?

No, but we can easily write something similar for creating one document per header, so that in a document that has headers like this: `h1 h2 h2 h3 h2 h3 h3 h2`, you would get following sections:

- `h1 h2 h2 h3 h2 h3 h3 h2`
- `h2`
- `h2 h3`
- `h3`
- `h2 h3 h3 h2`
- `h3`
- `h3`
- `h2`

The algorithm looks like this: visit all `headings` and for each heading create a new section. For each section, remove all nodes that come before the heading. Also remove the next heading that has a level equal or higher to the one selected and all following nodes.

The question is how you would integrate this. The index would get populated by the bot, so the bot might want to run a `plaintext` pipeline that extracts the sections as described above.
